Library/Formula/fsharp.rb

```diff
@@ -21,25 +21,60 @@ def install
     system "make", "install"
   end
 
+  def symlink real_file, link_file
+    system "ln", *["-s", real_file, link_file]
+  end
+
+  # handler takes arguments:
+  #  - tree_dir: the 'convoluted' non-absolute path the the installation, inside mono's prefix
+  #  - source_dir: tree_dir, inside mono's prefix, expanded to a full path
+  #  - fsharp_targ: the target file (for xbuild)
+  def iter_targets &handler
+    mono_ver = Formula["mono"].version
+    %w|3.0 3.1|.each do |fsharp_ver|
+      %w|Microsoft.Portable.FSharp.Targets
+         Microsoft.FSharp.Targets|.each do |fsharp_targ|
+        tree_dir   = "lib/mono/Microsoft\ SDKs/F\#/#{fsharp_ver}/Framework/v4.0"
+        source_dir = File.expand_path "#{prefix}/../../mono/#{mono_ver}/#{tree_dir}"
+        handler.call tree_dir, source_dir, fsharp_targ
+       end
+    end
+  end
```

Comment:
I disagree; `formula` files are not really software. We don't unit test individual methods in formula files etc.